feat: adding maven wrapper to project

rtroilo opened the pull request:
### Pull Request Checklist
<!--- Please make sure you have completed the following items BEFORE submitting a pull request (put an x in each box
when you have checked you have done them): -->
- [x] 1. I have [**rebased**][rebase] the latest version of the main branch into my feature branch and all conflicts
         have been resolved.
- [ ] 2. I have added information about the change/addition to functionality to the CHANGELOG.md file under the
         [Unreleased] heading.
- [ ] 3. I have documented my code using JDocs tags.
- [ ] 4. I have removed unnecessary commented out code, imports and System.out.println statements.
- [ ] 5. I have written JUnit tests for any new methods/classes and ensured that they pass.
- [ ] 6. I have created API tests for any new functionality exposed to the API.
- [ ] 7. If changes/additions are made to the ors-config.json file, I have added these to the [ors config documentation][config]
         along with a short description of what it is for, and documented this in the Pull Request (below).
- [ ] 8. I have built graphs with my code of the Heidelberg.osm.gz file and run the api-tests with all test passing
- [ ] 9. I have referenced the Issue Number in the Pull Request (if the changes were from an issue).
- [ ] 10. For new features or changes involving building of graphs, I have tested on a larger dataset
          (at least Germany), and the graphs build without problems (i.e. no out-of-memory errors).
- [ ] 11. For new features or changes involving the graphbuilding process (i.e. changing encoders, updating the
          importer etc.), I have generated longer distance routes for the affected profiles with different options
          (avoid features, max weight etc.) and compared these with the routes of the same parameters and start/end
          points generated from the current live ORS.
          If there are differences then the reasoning for these **MUST** be documented in the pull request.
- [ ] 12. I have written in the Pull Request information about the changes made including their intended usage
          and why the change was needed.
- [ ] 13. For changes touching the API documentation, I have tested that the API playground [renders correctly][api].

Fixes #1963 .

### Information about the changes
- Key functionality added:
- Reason for change:

### Examples and reasons for differences between live ORS routes, and those generated from this pull request
-

### Required changes to ors config (if applicable)
-

[config]: https://GIScience.github.io/openrouteservice/run-instance/configuration/
[api]: https://gitlab.gistools.geog.uni-heidelberg.de/giscience/openrouteservice-infrastructure/ors-docs-api#test-new-ors-documentation
[rebase]: https://github.com/GIScience/openrouteservice/blob/main/CONTRIBUTE.md#pull-request-guidelines

rtroilo committed:
feat: adding maven wrapper to project

koebi commented:
Adding a maven wrapper would change the way the project builds from source, no?
Should the docs be adapted to reflect that?

koebi commented:
Same for workflows and the dockerfile.

koebi commented:
From my POV, the docs should definitely reflect that.
The maven wrapper should also be used in the workflows, since they don't specify a `mvn` version at all.

The Dockerfile can stay on using `mvn` directly, since its version is controlled by the used image.

aoles reviewed (changes requested):
Thanks for your contribution @rtroilo , very much appreciated! ❤️ 

Do you think you could address the issues pointed out by @koebi, namely adapting the docs and workflows?

Cheers!

koebi commented:
Hey,
a second comment:
`mvnw` is used on Linux, `mvnw.cmd` is used on Windows.

Would we want to add (and thus support) both?
AFAIK none of our core devs is using Windows, and I don't know whether we can (or want) to "guarantee" that things will also work using Windows.

@takb @aoles do you have an opinion on that?

rtroilo commented:
> Would we want to add (and thus support) both? AFAIK none of our core devs is using Windows, and I don't know whether we can (or want) to "guarantee" that things will also work using Windows.

we could remove `mvnw.cmd` but it is just the wrapper for windows, so on windows machines, you could, if maven installed, as well do `mvn`. But I don't mind to remove the windows maven wrapper from this pr. 
you decide.

rtroilo commented:
> From my POV, the docs should definitely reflect that. The maven wrapper should also be used in the workflows, since they don't specify a `mvn` version at all.

I would say it makes sense, as we now would provide the wrapper with openrouteservice.
Would you change they docu or how should we proceed.

aoles commented:
@rtroilo It would be really great if you could make the adjustments mentioned in the review. In case you need any assistance or help please let us know. Cheers!

koebi commented:
> > Would we want to add (and thus support) both? AFAIK none of our core devs is using Windows, and I don't know whether we can (or want) to "guarantee" that things will also work using Windows.
> 
> we could remove `mvnw.cmd` but it is just the wrapper for windows, so on windows machines, you could, if maven installed, as well do `mvn`. But I don't mind to remove the windows maven wrapper from this pr. you decide.

I think I'm fine keeping `mvnw.cmd` for the moment :)

koebi commented:
Heyho,

> From my POV, the docs should definitely reflect that. The maven wrapper should also be used in the workflows, since they don't specify a `mvn` version at all.

I did a bit of digging, and if I'm not mistaken, the following files would have to be touched:
* `docs/technical-details/integration-tests.md`
* `docs/run-instance/building-from-source.md`

Everything under `docs/versions/…` does not need to be touched.

koebi committed:
docs: mention and use maven wrapper

MichaelsJP committed:
build: update Dockerfile to use Maven wrapper for dependency management

This changes it for the default docker environment as well as for the ors-test-scenario maven container

MichaelsJP committed:
docs: update the remaining documentation to use the Maven wrapper

Update the docs for api-benchmarks.md, coordinate-generators.md and the release instructions in RELEASE.md

MichaelsJP committed:
refactor: clean up old comment in pom.xml

MichaelsJP committed:
build: update the workflows to use Maven wrapper for building ors

MichaelsJP committed:
build: update ContainerInitializer to use Maven wrapper for running Maven commands

MichaelsJP commented:
Ready

MichaelsJP committed:
docs: add entry for Maven wrapper in changelog

MichaelsJP committed:
Merge branch 'main' into maven-wrapper

aoles committed:
Merge branch 'main' into maven-wrapper

aoles reviewed (approved):
Many thanks @MichaelsJP for finishing up the PR! ❤️
